egs/wsj/s5/steps/dict/train_g2p.sh

```diff
@@ -44,8 +44,10 @@ mkdir -p $wdir/log
 
 # Optionally remove words that are mapped to a single silence phone from the lexicon.
 if $only_words && [ ! -z "$silence_phones" ]; then
-  awk 'NR==FNR{a[$1] = 1; next} {s=$2;for(i=3;i<=NF;i++) s=s" "$i;a[$1]=s;if(!(s in a)) print $1" "s}' \
-    $silence_phones > $wdir/lexicon_onlywords.txt
+  awk -v w=$silence_phones \
+    'BEGIN{while((getline<w)>0) {for(i=1;i<=NF;i++) sil[$i]=1;}}
+    { p=$2; for(i=3;i<=NF;i++) p=p" "$i; 
+      if(!(p in sil)) print $1" "p }' $lexicon > $wdir/lexicon_onlywords.txt
```

Comment:
couldn't this be simplified to: `{ if (!(NF == 2 && $2 in sil)) print; }`
and I'd prefer to rename w to s.

Reply:
agreed. otherwise looks good. Thanks Vimal!